install/gcp/deployment_manager/istio-cluster.jinja.schema

```diff
@@ -0,0 +1,83 @@
+info:
+  title: GKE cluster
+  author: Google, Inc.
+  description: |
+    Creates a GKE cluster with a VM initializing istio.io for use within the cluster
+
+required:
+- gkeClusterName
+- zone
+- initialNodeCount
+- instanceType
+
+properties:
+  gkeClusterName:
+    type: string
+    description: GKE Cluster Name
+    default: istio-cluster
+  zone:
+    type: string
+    description: Zone in which the cluster should run.
+    default: us-central1-a
+    x-googleProperty:
+      type: GCE_ZONE
+  initialNodeCount:
+    type: integer
+    description: Initial number of nodes desired in the cluster.
+    default: 4
+    enum:
+      - 1
+      - 2
+      - 3
+      - 4
+      - 5
+      - 6
+  instanceType:
+    type: string
+    description: Node machineType
+    default: n1-standard-2
+    x-googleProperty:
+      type: GCE_MACHINE_TYPE
+      zoneProperty: zone
+
+  installIstioRelease:
+    type: string
+    description: Install Istio Release version.
+    default: 0.2.12
```

Comment:
how is this list going to be maintained ? it should probably be substituted from istio.VERSION ?
(or pulled from somewhere, like the release page, if it's a list)

Reply:
At the moment, it's community-maintained (which basically means someone will have to notice its out of date and update it - not ideal).

Agree that it should be pulled from some other source (although given the potential complexity of having to build the template, I'd prefer it in a followup-PR). What would you recommend?

Reply:
I'm a bit confused how this would all work:

Where are those files supposed to end up / how do they get packaged ?
The current release process only includes a small subset of yaml and readme/samples files to be part of the release, you would have to add those if it's meant to be consumed by end users (but isn't the whole point of the automatic installation that you don't need anything ?)

If they would be part of the release, they should be templatized like the other yaml (see istio-auth.yaml - it's generated by updateVersion.sh)

Reply:
Sorry, I didn't provide enough context.

This should probably not be packaged as a release, as you mentioned. Right now people can deploy it without downloading by specifying a url on the command line:

```
$ gcloud deployment-manager deployments create test16 --template https://raw.githubusercontent.com/istio/istio/install/[etc]
```

In the future we anticipate other methods of sourcing this template.